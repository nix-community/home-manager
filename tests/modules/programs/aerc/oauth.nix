{ config, pkgs, ... }: {
  config = {
    nmt.script = let
      dir = if (pkgs.stdenv.isDarwin && !config.xdg.enable) then
        "home-files/Library/Preferences/aerc"
      else
        "home-files/.config/aerc";
    in ''
      assertFileContent ${dir}/accounts.conf ${./oauth.expected}
    '';
    programs.aerc = {
      enable = true;
      extraConfig.general.unsafe-accounts-conf = true;
    };

    accounts.email.accounts = {
      basic = {
        realName = "Annie X. Hacker";
        userName = "anniex";
        address = "anniex@mail.invalid";
        primary = true;
        flavor = "outlook.office365.com";

        aerc = rec {
          enable = true;
          imapAuth = "xoauth2";
          smtpAuth = imapAuth;
          imapOauth2Params = {
            client_id = "9e5f94bc-e8a4-4e73-b8be-63364c29d753";
            token_endpoint =
              "https://login.microsoftonline.com/common/oauth2/v2.0/token";
          };
          smtpOauth2Params = imapOauth2Params;
        };
      };
    };
  };
}

