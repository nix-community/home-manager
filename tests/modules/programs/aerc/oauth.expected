# Generated by Home Manager.

[basic]
copy-to = Sent
default = Inbox
from = Annie X. Hacker <anniex@mail.invalid>
outgoing = smtp+xoauth2://anniex@smtp.office365.com:587?client_id=9e5f94bc-e8a4-4e73-b8be-63364c29d753&token_endpoint=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Ftoken
postpone = Drafts
source = imaps+xoauth2://anniex@outlook.office365.com:993?client_id=9e5f94bc-e8a4-4e73-b8be-63364c29d753&token_endpoint=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Ftoken
